sql/catalyst/src/main/scala/org/apache/spark/sql/connector/catalog/V1Table.scala

```diff
@@ -78,6 +78,8 @@ private[sql] case class V1Table(v1Table: CatalogTable) extends Table {
       partitions += spec.asTransform
     }
 
+    import org.apache.spark.sql.connector.catalog.CatalogV2Implicits.TransformHelper
+    partitions.validatePartitionColumns()
```

Comment:
I agree that v1 tables should throw an exception if nested columns are used because it wasn't supported in v1, but I think this change is over-broad. There's no need to disallow nested columns in all v2 sources.

Reply:
Digging the v2 codebase a bit, and I feel currently, we need some work to actually get v2 sources supporting using nested columns in transform. For example, in `FileTable`, `override def partitioning: Array[Transform]` is converted from `PartitionSpec` which doesn't support nested column at all. Since we are mixing v1 and v2 code here and there, do we have a plan to untangle them so it will be easier to extend nested column support in v2?

Reply:
FileTable doesn't support partitioning by nested columns, and that's okay because it is optional. Sources should just reject partitioning that is not supported.